600.java

```diff
@@ -0,0 +1,95 @@
+x_1,x_2,...,x_N
+where 1 <= N <= 100,000 and the sequence (x_1, x_2,..., x_N) is a permutation of the set of integers {1, 2,..., N} (i.e., the sequence is not empty, and there are no repeated or missing values).
+Your program should produce exactly one line of output, a comma-separated list of operations with no spaces in the following format:
+op_1,op_2,...,op_2N
+where each op_i is a string from the set {pushFront, pushBack, popBack}. The output of the sequence of operations (op_1, op_2,..., op_2N) should be the sequence (x_1, x_2,..., x_N). If it is not possible to produce the output (x_1, x_2,..., x_N) with any sequence of single-output deque operations, simply print the string “impossible”. If there are multiple sequences of operations that result in the sequence received in the input, choose the output that is smallest lexicographically, ordered by standard alphabetical ordering.
+
+Sample Testcases
+Input:
+3,2,1
+Output:
+pushBack,pushBack,pushBack,popBack,popBack,popBack
+
+Input:
+1,2,3
+Output (note choice of pushBack over pushFront):
+pushBack,popBack,pushBack,popBack,pushBack,popBack
+
+Input:
+4,1,5,2,3
+Output (pushFront is needed in some cases):
+pushBack,pushFront,pushFront,pushBack,popBack,popBack,pushBack,popBack,popBack,popBack
+
+Input:
+5,1,4,2,3
+Output (some sequences are impossible):
+impossible
+
+To help with solving this problem in a timely manner, we provide the following hints that may help you work out one way of writing an efficient program to solve this problem:
+1. Note that if the first element of the input sequence is 4, then the 5th element of the output must be popBack (unless the output is impossible). The reason for this is that at least 4 pushes must be executed to get 4 into the deque, and the 4th push must be a pushBack so that 4 is ready to be popped from the back of the deque. More than 4 pushes could be executed, but elements 5 and above must be pushed to the front, and this could easily be done after 4 was popped instead of before. Since popBack is lexicographically smaller than pushFront, we prefer to execute popBack as early as possible.
+2. Consider simulating the deque as a way to efficiently determine which operations were performed on it. For example, as above, if the first element of the input is a 4, simulate a deque having the elements 1 through 4 pushed into it. Since you do not know whether each element was pushed to the front or the back, try pushing it on both sides and figuring out which side is correct later in the simulation.
+**/
+
+public static void main(String[] args){
+	Scanner sc = new Scanner(System.in);
+	String input = sc.nextLine();
+	String[] strs = strs.split(",");
+	int n = strs.length;
+	int[] nums = new int[n];
+	for(int i = 0; i < n; i++){
+		nums[i] = Integer.parseInt(strs[i]);
+	}
+	int[] order = new int[n];
+	for(int i = 0; i < n; i++){
+		order[nums[i]] = i;
+	}
+	Deque<Integer> deque = new LinkedList<>();
+	StringBuilder sb = new StringBuilder();
```

Comment:
what does sb stand for? String builder? Instead of repeating its type and giving no meaning for this variable, rename it!